test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/internal/CreateMockitoMocksCallback.java

```diff
@@ -0,0 +1,70 @@
+package io.quarkus.test.junit.mockito.internal;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Qualifier;
+
+import org.mockito.Mockito;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.test.junit.callback.QuarkusTestBeforeAllCallback;
+import io.quarkus.test.junit.mockito.MockBean;
+
+public class CreateMockitoMocksCallback implements QuarkusTestBeforeAllCallback {
+
+    @Override
+    public void beforeAll(Object testInstance) {
+        Class<?> current = testInstance.getClass();
+        while (current.getSuperclass() != null) {
+            for (Field field : current.getDeclaredFields()) {
+                MockBean mockBeanAnnotation = field.getAnnotation(MockBean.class);
+                if (mockBeanAnnotation != null) {
+                    Object beanInstance = getBeanInstance(testInstance, field);
+                    Object mock = createMockAndSetTestField(testInstance, field, beanInstance);
+                    MockitoMocksTracker.track(testInstance, mock, beanInstance);
+                }
+            }
+            current = current.getSuperclass();
+        }
+    }
+
+    private Object createMockAndSetTestField(Object testInstance, Field field, Object beanInstance) {
+        Object mock = Mockito.mock(beanInstance.getClass());
+        field.setAccessible(true);
+        try {
+            field.set(testInstance, mock);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        return mock;
+    }
+
+    private Object getBeanInstance(Object testInstance, Field field) {
+        Class<?> fieldClass = field.getType();
+        InstanceHandle<?> instance = Arc.container().instance(fieldClass, getQualifiers(field));
```

Comment:
If you store the `InstanceHandle` you can use it later to destroy a `@Dependent` bean correctly. See `io.quarkus.arc.InstanceHandle.close()`.

Reply:
We can't really mock @Dependent beans anyway.

Reply:
Hm... Here we aren't storing the references, just using them to check and then obtain the bean. So we don't need to close them anywhere, no?

Reply:
> We can't really mock @dependent beans anyway.

Ah, that's correct. For `@Dependent` it would fail later when we try to install the mock by means of `QuarkusMock`.

> So we don't need to close them anywhere, no?

It's ok for normal scoped beans and `@Singleton`. Nok for `@Dependent`.

Reply:
Ah, now I see what your concern was! So yeah, no `@Dependent` - no problem :)